api/metadata/models.py

```diff
@@ -0,0 +1,126 @@
+from urllib.parse import urlparse
+
+from core.models import AbstractBaseExportableModel
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from organisations.models import Organisation
+from projects.models import Project
+
+from .fields import GenericObjectID
+
+FIELD_VALUE_MAX_LENGTH = 2000
+
+METADATA_SUPPORTED_MODELS = ["environment"]
+
+# A map of model name to a function that takes the object id and returns the organisation_id
```

Comment:
I'm not sure why we have two levels here? What relevance does the `"environment"` top level key have?

Reply:
I guess this is helping somehow when we want to say that 'metadata field x is required for this project / organisation' ?